include/deal.II/dofs/dof_handler.h

```diff
@@ -418,19 +438,38 @@ class DoFHandler : public Subscriptor
    * finite element indices may be used on different cells, and the default
    * index there corresponds to an invalid value.
    */
-  static const unsigned int default_fe_index = 0;
+  static const unsigned int default_fe_index =
+    numbers::invalid_unsigned_int; // TODO
+
+  /**
+   * The type in which we store the active FE index.
+   */
+  using active_fe_index_type = unsigned short int;
+
+  /**
+   * The type in which we store the offsets in the CRS data structures.
+   */
+  using offset_type = unsigned int;
```

Comment:
Are these type fine? `unsigned int`?

Reply:
This limits us to 4G local dofs on the local MPI rank, or, more precisely, 4G indices accumulated over all cells? I am fine with that, since there is a `typedef` that would allow us to change once local memory has become sufficient to make this case more common. It might be a while until that happens because progress in terms of memory sizes or dollar/GB has slowed down quite a bit during the last five years.